src/compiler/transformers/es2015.ts

```diff
@@ -915,14 +915,14 @@ namespace ts {
                 // The assumption is that no prior step in the pipeline has added any prologue directives.
                 statementOffset = 0;
             }
-            else if (constructor) {
-                // Otherwise, try to emit all potential prologue directives first.
-                statementOffset = addPrologueDirectives(statements, constructor.body.statements, /*ensureUseStrict*/ false, visitor);
-            }
 
             if (constructor) {
                 addDefaultValueAssignmentsIfNeeded(statements, constructor);
                 addRestParameterIfNeeded(statements, constructor, hasSynthesizedSuper);
+                if (!hasSynthesizedSuper) {

```

Comment:
Maybe we need to split `addPrologueDirectives` into two functions, `addPrologueDirectives` (which only handles *real* prologue directives) and `addCustomPrologueDirectives` (which only handles our custom leading statements).